### PR TITLE
Allow EnumNode name to be null

### PR DESCRIPTION
**Pull request by MatTheCat**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #25046
| License       | MIT

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Thanks, could you add a test case please?

**Comment by MatTheCat** [comment]
Yep, guess 9986c5e will be enough?

**Review by xabbuh — APPROVED** [review]

File: src/Symfony/Component/Config/Tests/Definition/EnumNodeTest.php [hunk 1]
```diff
@@ -43,6 +43,11 @@ public function testConstructionWithOneDistinctValue()
         $this->assertSame('foo', $node->finalize('foo'));
     }
 
+    public function testConstructionWithNullName()
+    {
+        new EnumNode(null, null, array('foo'));
```
Comment (chalasr) [inline src/Symfony/Component/Config/Tests/Definition/EnumNodeTest.php]: please call `->finalize()` or so to make the test not risky

File: src/Symfony/Component/Config/Tests/Definition/EnumNodeTest.php [hunk 2]
```diff
@@ -43,6 +43,12 @@ public function testConstructionWithOneDistinctValue()
         $this->assertSame('foo', $node->finalize('foo'));
     }
 
+    public function testConstructionWithNullName()
+    {
+        $node = new EnumNode(null, null, array('foo'));
+        $node->finalize('foo');
```
Comment (chalasr) [inline src/Symfony/Component/Config/Tests/Definition/EnumNodeTest.php]: please assert it (sorry I've been unclear) :)

**Comment by MatTheCat** [comment]
Woops sorry :sweat:

**Review by chalasr — APPROVED** [review]

**Review by stof — APPROVED** [review]

**Review by ogizanagi — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @MatTheCat.

**Pull request closed without merging**